pkg/inference/backends/llamacpp/llamacpp.go

```diff
@@ -213,3 +215,50 @@ func (l *llamaCpp) GetDiskUsage() (int64, error) {
 	}
 	return size, nil
 }
+
+func (l *llamaCpp) GetRequiredMemoryForModel(model string, config *inference.BackendConfiguration) (*inference.RequiredMemory, error) {
+	mdl, err := l.modelManager.GetModel(model)
+	if err != nil {
+		return nil, err
+	}
+	mdlPath, err := mdl.GGUFPath()
+	if err != nil {
+		return nil, err
+	}
+	mdlGguf, err := parser.ParseGGUFFile(mdlPath)
+	if err != nil {
+		return nil, err
+	}
+	mdlConfig, err := mdl.Config()
+	if err != nil {
+		return nil, err
+	}
+
+	contextSize := GetContextSize(&mdlConfig, config)
+
+	// FIXME(p1-0tr): for now assume we are running on GPU (single one) - Devices[1];
+	// sum up weights + kv cache + context for an estimate of total GPU memory needed
+	// while running inference with the given model
+	estimate := mdlGguf.EstimateLLaMACppRun(parser.WithLLaMACppContextSize(int32(contextSize)),
+		// FIXME(p1-0tr): add logic for resolving other param values, instead of hardcoding them
+		parser.WithLLaMACppLogicalBatchSize(2048),
+		parser.WithLLaMACppOffloadLayers(100)) // FIXME(p1-0tr): need to set this to 0 for CPU only case
```

Comment:
This `FIXME` still needs to be resolved (it's slightly annoying because we don't have a straightforward way of telling if we are running with or without GPU acceleration at this point)

Reply:
That or check for the NVIDIA environment variables that the CTK sets, e.g. `NVIDIA_VISIBLE_DEVICES`